pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,140 @@
+	}
+	if networkName == "" {
+		// no network was provided, this will be the name
+		networkName = fmt.Sprintf("%s-network", clusterID.InfraID)
+	}
+
+	// auto create subnets is not used as the subnet information is provided through
+	// the install config
+	autoCreateSubnets := false
+	enableFlowLogs := os.Getenv("GCP_ENABLE_FLOW_LOGS") == "true"
+
+	masterSubnet := installConfig.Config.GCP.ControlPlaneSubnet
+	masterSubnetDescription := "Control Plane Subnet used by the installer"
+	if masterSubnet == "" {
+		// TODO: common function - this will be used in machine manifest creation too
+		masterSubnet = fmt.Sprintf("%s-%s-subnet", clusterID.InfraID, "master")
+		masterSubnetDescription = "Control Plane Subnet owned by the installer"
+	}
+	master := capg.SubnetSpec{
+		Name:           masterSubnet,
+		CidrBlock:      "",
+		Description:    &masterSubnetDescription,
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: &enableFlowLogs,
+		Purpose:        &masterSubnetDescription,
+	}
+
+	workerSubnet := installConfig.Config.GCP.ComputeSubnet
+	workerSubnetDescription := "Compute Subnet used by the installer"
+	if workerSubnet == "" {
+		// TODO: common function - this will be used in machine manifest creation too
+		workerSubnet = fmt.Sprintf("%s-%s-subnet", clusterID.InfraID, "worker")
+		workerSubnetDescription = "Compute Subnet owned by the installer"
+	}
+	worker := capg.SubnetSpec{
+		Name:           workerSubnet,
+		CidrBlock:      "",
+		Description:    &workerSubnetDescription,
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: &enableFlowLogs,
+		Purpose:        &workerSubnetDescription,
+	}
+
+	// Add the CIDR information
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, _ := net.ParseCIDR(machineV4CIDRs[0])
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %s", err.Error())
+		}
+		// The subnet must be created, provide the CIDR
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %s", err.Error())
+		}
+		// The subnet must be created, provide the CIDR
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := make(map[string]string, len(installConfig.Config.GCP.UserLabels)+1)
+	// add OCP default label
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
```

Comment:
Is a featurecheck needed on using these labels since installconfig has a comment:
// This is a TechPreview feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags featureGate
// enabled or TechPreviewNoUpgrade featureSet to configure labels.

Reply:
The checks should be either completed in types or the install config validation.